tests/csharp-stellar-base.Tests/AssetTests.cs

```diff
@@ -0,0 +1,80 @@
+        {
+            Asset asset = new Asset();
+
+            Assert.AreEqual(Asset.AssetTypeEnum.ASSET_TYPE_NATIVE, asset.Type);
+
+            Stellar.Generated.Asset genAsset = asset.ToXDR();
+
+            Assert.AreEqual(Stellar.Generated.AssetType.AssetTypeEnum.ASSET_TYPE_NATIVE, genAsset.Discriminant.InnerValue);
+
+            Asset resAsset = Asset.FromXDR(genAsset);
+
+            Assert.AreEqual(Asset.AssetTypeEnum.ASSET_TYPE_NATIVE, resAsset.Type);
+        }
+
+        [TestMethod]
+        public void TestAlphaNum4Asset()
+        {
+            var keyPair = KeyPair.Master();
+            string code = "Test";
+            Asset asset = new Asset(code, keyPair);
+
+            Assert.AreEqual(code, asset.Code);
+            Assert.AreEqual(keyPair, asset.Issuer);
+            Assert.AreEqual(Asset.AssetTypeEnum.ASSET_TYPE_CREDIT_ALPHANUM4, asset.Type);
+
+            Stellar.Generated.Asset genAsset = asset.ToXDR();
+
+            Assert.AreEqual(Encoding.ASCII.GetBytes(code).ToString(), genAsset.AlphaNum4.AssetCode.ToString());
+            Assert.AreEqual(keyPair.PublicKey.ToString(), genAsset.AlphaNum4.Issuer.InnerValue.Ed25519.InnerValue.ToString());
+            Assert.AreEqual(Stellar.Generated.AssetType.AssetTypeEnum.ASSET_TYPE_CREDIT_ALPHANUM4, genAsset.Discriminant.InnerValue);
+
+            Asset resAsset = Asset.FromXDR(genAsset);
+
+            Assert.AreEqual(code, resAsset.Code);
+            Assert.AreEqual(keyPair.Address, resAsset.Issuer.Address, keyPair.Address);
+            Assert.AreEqual(keyPair.PublicKey.ToString(), resAsset.Issuer.PublicKey.ToString());
+            Assert.AreEqual(Asset.AssetTypeEnum.ASSET_TYPE_CREDIT_ALPHANUM4, resAsset.Type);
+        }
+
+        [TestMethod]
+        public void TestAlphaNum12Asset()
+        {
+            var keyPair = KeyPair.Master();
+            string code = "TestTestTest";
+            Asset asset = new Asset(code, keyPair);
+
+            Assert.AreEqual(code, asset.Code);
+            Assert.AreEqual(keyPair, asset.Issuer);
+            Assert.AreEqual(Asset.AssetTypeEnum.ASSET_TYPE_CREDIT_ALPHANUM12, asset.Type);
+
+            Stellar.Generated.Asset genAsset = asset.ToXDR();
+
+            Assert.AreEqual(Encoding.ASCII.GetBytes(code).ToString(), genAsset.AlphaNum12.AssetCode.ToString());
+            Assert.AreEqual(keyPair.PublicKey.ToString(), genAsset.AlphaNum12.Issuer.InnerValue.Ed25519.InnerValue.ToString());
+            Assert.AreEqual(Stellar.Generated.AssetType.AssetTypeEnum.ASSET_TYPE_CREDIT_ALPHANUM12, genAsset.Discriminant.InnerValue);
+
+            Asset resAsset = Asset.FromXDR(genAsset);
+
+            Assert.AreEqual(code, resAsset.Code);
+            Assert.AreEqual(keyPair.Address, resAsset.Issuer.Address);
+            Assert.AreEqual(keyPair.PublicKey.ToString(), resAsset.Issuer.PublicKey.ToString());
+            Assert.AreEqual(Asset.AssetTypeEnum.ASSET_TYPE_CREDIT_ALPHANUM12, resAsset.Type);
+        }
+    }
```

Comment:
I'm missing the unhappy path tests here. What if code == null, keyPair == null etc